SalesforceCommerceCloud/pwa-kit
# @W-16909794 - Add passwordless/social login UI buttons to Checkout page

# Description
To enable social and passwordless login in the PWA Kit, we need to ensure that these login methods are also available in the Checkout flow. Today, users can either check out as a **GUEST** or log in with an **email and password**.

If a customer has enabled social and/or passwordless login, we want to add these as login options.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Modify the `<ContactInfo>` component that is rendered in the Checkout page
- If Social and Passwordless Login are enabled, render these login buttons
- If they are **NOT** enabled, maintain the original log in flow (guest or credentials)

# How to Test-Drive This PR

- Enable both social and passwordless login in the config
- Verify that the ContactInfo card looks as follows:
<img width="500" alt="Screenshot 2024-10-31 at 12 14 35 PM" src="https://github.com/user-attachments/assets/e7a5e845-68d8-40db-b54b-1be95e48ac6a">

- When user logs in with password, the other buttons should disappear:
https://github.com/user-attachments/assets/ad917723-82df-4ebe-b6a8-253ca6c98a65

- Disable both social and passwordless login in the config
- Verify that the ContactInfo card looks as follows:
<img width="500" alt="Screenshot 2024-10-31 at 12 15 07 PM" src="https://github.com/user-attachments/assets/dc9d276d-d384-4101-b08a-fb1f5f4e9210">


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
